app/src/test/java/com/meewii/rateconverter/business/ExchangeRateRepositoryTest.kt

```diff
@@ -276,15 +210,27 @@ class ExchangeRateRepositoryTest {
     testSubscriber.assertValue { (it as RateList.Success).currencies[2].calculatedValue == 4.0 }
   }
 
-  //TODO change behavior of this case
   @Test
-  fun `get combined rates, error from API, after 16`() {
+  fun `get combined rates, error from API, success from DB, should return DB data`() {
     // having
-    val now = LocalDateTime.of(2019, 3, 7, 17, 0)
-    clockMock = Clock.fixed(now.atZone(ZoneId.systemDefault()).toInstant(), ZoneId.systemDefault())
-    sut = ExchangeRateRepository(serviceMock, daoMock, clockMock, userInputRepositoryMock, pinedCurrenciesRepositoryMock)
-
     whenever(serviceMock.getRatesForBase(any())).thenReturn(Single.error(TestException("error")))
+    whenever(daoMock.getRatesForBase(any())).thenReturn(Single.just(entity))
+    whenever(userInputRepositoryMock.getUserInputsStream()).thenReturn(Flowable.just(1.0))
+    whenever(pinedCurrenciesRepositoryMock.getPinedCurrencies()).thenReturn(Flowable.just(setOf()))
+
+    // when
+    val testSubscriber = sut.getCombinedRates("GBP").test()
+    testSubscriber.awaitTerminalEvent()
+
+    // then
+    testSubscriber.assertValue(successDbRateList)
+  }
+
+  @Test
```

Comment:
Instead of asserting error, I personally love @Test(expected = TextException.class). This way I spare myself one additional line of code :D :D